boa/src/builtins/function/mod.rs

```diff
@@ -28,7 +28,7 @@ use gc::{unsafe_empty_trace, Finalize, Trace};
 use std::fmt::{self, Debug};
 
 /// _fn(this, arguments, ctx) -> ResultValue_ - The signature of a built-in function
-pub type NativeFunctionData = fn(&mut Value, &[Value], &mut Interpreter) -> ResultValue;
+pub type NativeFunctionData = fn(&Value, &[Value], &mut Interpreter) -> ResultValue;
```

Comment:
Whats the reason behind the change here?

Reply:
> Whats the reason behind the change here?

It's because the `Value` does not need to be mutable itself, except `Object` but object is covered in a `GcCell<...>` it has interior mutability, so we can `borrow_mut` the `Object` with a immutable reference to `Value`.

This eliminated some places where we were awkwardly cloning the value and passing it to a function because the builtin function required it (`&mut value.clone()`), instead of just passing the value direclty.